test/metabase/search/fulltext/scoring_test.clj

```diff
@@ -0,0 +1,107 @@
+(ns metabase.search.fulltext.scoring-test
+  (:require
+   [clojure.test :refer :all]
+   ;; For now, this is specialized to postgres, but we should be able to abstract it to all index-based engines.
+   [metabase.search :as search]
+   [metabase.search.config :as search.config]
+   [metabase.search.impl :as search.impl]
+   [metabase.search.postgres.core :as search.postgres]
+   [metabase.search.postgres.index :as search.index]
+   [metabase.search.postgres.ingestion :as search.ingestion]
+   [metabase.search.postgres.scoring :as search.scoring]
+   [metabase.test :as mt])
+  (:import (java.time Instant)
+           (java.time.temporal ChronoUnit)))
+
+(set! *warn-on-reflection* true)
+
+;; arbitrary
+(def ^:private user-id 7331)
+
+;; We act on a random localized table, making this thread-safe.
+#_{:clj-kondo/ignore [:metabase/test-helpers-use-non-thread-safe-functions]}
+(defmacro with-index-contents
+  "Populate the index with the given appdb agnostic entity shapes."
+  [entities & body]
+  `(when (search/supports-index?)
+     (let [table-name# (search.index/random-table-name)]
+       (try
+         (binding [search.index/*active-table* table-name#]
+           (search.index/create-table! table-name#)
+           (#'search.index/batch-upsert! table-name#
+                                         ;; yup, we have two different shapes called "entry" at the moment
+                                         (map (comp #'search.index/entity->entry
+                                                    #'search.ingestion/->entry)
+                                              ~entities))
+           ~@body)
+         (finally
+           (#'search.index/drop-table! table-name#))))))
+
+(defn- search [ranker-key search-string & {:as raw-ctx}]
+  (let [search-ctx (search.impl/search-context
+                    (merge
+                     {:archived              false
+                      :search-string         search-string
+                      :current-user-id       user-id
+                      :current-user-perms    #{"/"}
+                      :is-superuser?         true
+                      :is-impersonated-user? false
+                      :is-sandboxed-user?    false
+                      :models                search.config/all-models
+                      :model-ancestors?      false}
+                     raw-ctx))]
+    (is (get (search.scoring/scorers search-ctx) ranker-key) "The ranker is enabled")
+    (map (juxt :model :id :name) (#'search.postgres/fulltext search-string search-ctx))))
+
+;; ---- index-ony rankers ----
+;; These are the easiest to test, as they don't depend on other appdb state.
+
+(deftest ^:parallel model-test
+  (with-index-contents
```

Comment:
a context explaining the reasons why this is expected order would be nice for future, in case we change the weights.

Reply:
For now I've added `testing` explanations, but we can also add inline comments for more nuanced cases.